tests/functional/test_s3.py

```diff
@@ -170,14 +183,14 @@ def test_default_endpoint_for_virtual_addressing(self):
     def test_provided_endpoint_url_for_virtual_addressing(self):
         s3 = self.session.create_client(
             's3', config=Config(s3={'addressing_style': 'virtual'}),
-            endpoint_url='https://foo.amazonaws.com')
```

Comment:
We aren't likely to be passed an amazonaws address as a custom endpoint url. And in the case of this regex change, it could fail.